src/fsharp/FSharp.Core/list.fs

```diff
@@ -692,3 +692,51 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Unfold")>]
         let unfold<'T, 'State> (generator:'State -> ('T*'State) option) (state:'State) = Microsoft.FSharp.Primitives.Basics.List.unfold generator state
+
+        [<CompiledName("RemoveAt")>]
+        let removeAt (index: int) (source: 'T list) : 'T list =
+            if index < 0 || index >= source.Length then invalidArg "index" "index must be within bounds of the array"
```

Comment:
The `.Length` requires an iteration of the list.  I think it's non-optimal

Reply:
I rewrote all the list implementations to use ListCollector instead